fix(dmsquash-live): handle relative pathspec

LaszloGombos opened the pull request:
This change is required to make the following canonical example work that is mentioned in the [documentation](https://github.com/dracutdevs/dracut/blob/master/man/dracut.cmdline.7.asc):

> rd.live.overlay=/dev/sdb1:persistent-overlay.img

This example has been in the dracut documentation since [2015](https://github.com/dracutdevs/dracut/commit/a1b4efe6a722ac3754d8a48109575d11d5ccf61b).

At the same time, improve the documentation with a more specific example.

## Checklist
- [X] I have tested it locally
- [X] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Fixes #579 (partially)

CC @FGrose @Conan-Kudo @fabiand

aafeijoo-suse reviewed (commented):
> This change is required to make the following canonical example work that is mentioned in the documentation

Are we sure that the example in the man page (`pathspec` not starting with / ) is correct?

If the answer is yes, wouldn't it be easier to just fix when `pathspec` is assigned and doesn't start with / , to avoid the "//" case?

https://github.com/dracutdevs/dracut/blob/aca51203609954b38b41ea303a4376cbce061150/modules.d/90dmsquash-live/dmsquash-live-root.sh#L140

LaszloGombos committed:
fix(dmsquash-live): handle relative pathspec

This change is required to make the earlier canonical
example work that is mentioned in the documentation:

At the same time, improve the documentation with a more
specific example.

aafeijoo-suse commented on modules.d/90dmsquash-live/dmsquash-live-root.sh:
```diff
@@ -145,7 +145,7 @@ do_live_overlay() {
     elif strstr "$overlay" ":"; then
         # pathspec specified, extract
-        pathspec=${overlay##*:}
+        pathspec=/${overlay##*:}
```
I would only prepend / if `pathspec` does not start with /.
With your code, `rd.live.overlay=/dev/disk/by-uuid/XXXX:/path/to/overlay.img` ends with `pathspec=//path/to/overlay.img`

aafeijoo-suse reviewed (commented):
> > Are we sure that the example in the man page (`pathspec` not starting with / ) is correct?
> 
> I think the example in the documentation can be improve. I added the improvement in the doc to this PR as well.
> 
> Regardless this example has been in the documentation for 7 years, and maintaining compatibility with this example seems to be simple enough.
 
What I meant was I haven't found any example of `pathspec` without starting with /, because it refers to a file path.
